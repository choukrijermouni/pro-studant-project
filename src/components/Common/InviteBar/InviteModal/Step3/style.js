import {
  neutral
} from '@pro_boa/ui'

export default {
  modal: {
    width: 636,
    backgroundColor: neutral[0],
    borderRadius: 4
  },
  line: {
    backgroundColor: neutral[13],
    height: 1,
    border: 'none',
    marginBottom: 20,
    width: '90%'
  },
  image: {
    maxWidth: 283
  }
}
