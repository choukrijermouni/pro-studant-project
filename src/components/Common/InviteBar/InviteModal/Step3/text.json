{
    "part1": "Félicitation ",
    "part2": ", vos apprenants ont reçu une invitation par e-mail pour vous rejoindre",
    "button": "Ok"
}