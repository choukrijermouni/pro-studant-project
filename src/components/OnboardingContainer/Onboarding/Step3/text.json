{
  "header": "Modification du mot de passe.",
  "save": "Continuer",
  "passwordLabel": "Mot de passe",
  "confirmPasswordLabel": "Confirmation du mot de passe",
  "Bonjour": "Bonjour",
  "changePassword": " pour garantir la confidentialité et la sécurité de vos données, veuillez modifier votre mot de passe",
  "contact": "En cas de besoin, contactez notre équipe par téléphone au :",
  "phoneNumber": "+33 1 77 62 45 80",
  "contactEmail": "contact@pro_boa.com"
}