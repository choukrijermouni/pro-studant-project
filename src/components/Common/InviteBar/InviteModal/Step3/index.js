import {
  TypographyElement,
  Row,
  Button,
  ButtonVariation,
  ButtonSize
} from '@pro_boa/ui'
import { createUseStyles } from 'react-jss'
import style from './style'
import Text from './text.json'
import { congratulations } from 'assets'
import { useDispatch, useSelector } from 'react-redux'
import { updateLearnersAfterInviteAction } from 'pages/Learners/store'

const useStyles = createUseStyles(style)
export default ({ closeModal }) => {
  const dispatch = useDispatch()
  const { modal, line, image } = useStyles()
  const LastName = useSelector(({ identity }) => identity?.user?.family_name)
  return (
    <div>
      <div className={modal}>
        <img className={image} src={congratulations} alt='congratulations' />
        <TypographyElement
          textAlign='center'
          component='h1'
          align='center'
          variant='heading1'
          spacing='0 0 30px 0'
          style={{ textAlign: 'center' }}
        >
          {Text.part1}{LastName}{Text.part2}
        </TypographyElement>
        <hr className={line} />
        <Row className='row' justify='right'>
          <div className='content_2'>
            <Button
              dataTest='close-modal-button'
              variation={ButtonVariation.primary}
              size={ButtonSize.big}
              width={153}
              height={54}
              handleClick={() => {
                closeModal()
                dispatch(updateLearnersAfterInviteAction())
              }}
              label={Text.button}
            />
          </div>
        </Row>
      </div>
    </div>
  )
}
