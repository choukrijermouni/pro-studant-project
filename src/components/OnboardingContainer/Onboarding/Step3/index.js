import {
  Container,
  TypographyElement,
  Button,
  PasswordInput,
  Paper,
  yellow,
  Icon,
  Icons
} from '@pro_boa/ui'
import style from './style'
import Text from './text.json'
import { createUseStyles } from 'react-jss'
import { useFormik } from 'formik'
import { validatePassword } from 'helpers/validator'
import { editOrganizationProfilePasswordAction } from 'pages/MyAccount/store'
import { useDispatch } from 'react-redux'
import { logo as pro_boa } from 'assets/index'
import { onboardingSteps, salesB2B } from 'constants/'

const useStyle = createUseStyles(style)

const noNotifications = true

export default ({ setStep }) => {
  const dispatch = useDispatch()
  const {
    rootClass,
    horizantalDivider,
    passwordStyle,
    licenseButtons,
    body,
    paper,
    logo,
    content,
    contactCard,
    icon,
    contact
  } = useStyle()
  const { handleChange: onFormikHandleChange, handleSubmit, errors, touched, values } = useFormik({
    initialValues: {
      confirmPassword: '',
      password: ''
    },
    validate: validatePassword,
    onSubmit: values => {
      dispatch(editOrganizationProfilePasswordAction(values.password, noNotifications))
      setStep(onboardingSteps.success)
    }
  })

  return (
    <div className={body}>
      <Paper className={paper}>
        <div className={content}>
          <Container className={rootClass}>
            <img src={pro_boa} alt='notification' className={logo} />
            <TypographyElement
              component='h3'
              spacing='56px 0 16px 0'
              lineHeight='40px'
              variant='heading2'
              fontWeight='bold'
            >
              {Text.header}
            </TypographyElement>
            <TypographyElement
              component='h4'
              variant='heading4'
              align='left'
              spacing='8px 0 34px 0'
              fontWeight='bold'
              color={yellow[3]}
            >
              {`${Text.Bonjour}${Text.changePassword}`}
            </TypographyElement>
            <form onSubmit={handleSubmit}>
              <div className={passwordStyle}>
                <PasswordInput
                  id='password'
                  handleChange={onFormikHandleChange}
                  value={values.password}
                  type='password'
                  status={(errors.password && touched.password) && 'error'}
                  label={Text.passwordLabel}
                  placeholder={Text.passwordLabel}
                  message={errors.password}
                />
              </div>
              <PasswordInput
                id='confirmPassword'
                handleChange={onFormikHandleChange}
                value={values.confirmPassword}
                type='password'
                label={Text.confirmPasswordLabel}
                status={(errors.confirmPassword && touched.confirmPassword) && 'error'}
                placeholder={Text.confirmPasswordLabel}
                message={errors.confirmPassword}
              />
              <TypographyElement
                component='h4'
                variant='heading4'
                align='left'
                spacing='34px 0 16px 0'
                fontWeight='bold'
              >
                {Text.contact}
              </TypographyElement>
              <div className={contactCard}>
                <div className={contact}>
                  <Icon iconName={Icons.smartPhone} style={icon} />
                  <TypographyElement
                    variant='heading4'
                    align='left'
                    spacing='17px 0 16px 16px'
                    fontWeight='bold'
                    Component='a'
                    cursor='pointer'
                    href={`tel:${salesB2B.SalesPhone}`}
                  >
                    {salesB2B.SalesPhone}
                  </TypographyElement>
                </div>
                <div className={contact}>
                  <Icon iconName={Icons.envelope} style={icon} />
                  <TypographyElement
                    variant='heading4'
                    align='left'
                    spacing='17px 0 16px 16px'
                    fontWeight='bold'
                    Component='a'
                    cursor='pointer'
                    href={`mailto:${salesB2B.SalesEmail}`}
                  >
                    {salesB2B.SalesEmail}
                  </TypographyElement>
                </div>

              </div>
              <div className={horizantalDivider} />
              <div className={licenseButtons}>
                <Button
                  handleClick={() => handleSubmit()}
                  width={153}
                  label={Text.save}
                />
              </div>
            </form>
          </Container>
        </div>
      </Paper>
    </div>
  )
}
