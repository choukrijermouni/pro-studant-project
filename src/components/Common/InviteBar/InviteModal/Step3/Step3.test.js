import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import Step3 from '.'
configure({ adapter: new Adapter() })

describe('>>> Step3 --- Shallow Render Step3 component', () => {
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(<Step3 />)
    tree = renderer.create(
      <Step3 />
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
