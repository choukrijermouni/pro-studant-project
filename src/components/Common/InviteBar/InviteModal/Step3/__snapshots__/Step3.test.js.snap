// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Step3 --- Shallow Render Step3 component +++ render correctly and match snapshot 1`] = `
<div>
  <div
    className="modal-0-2-1"
  >
    <img
      alt="thankYou"
      src={Object {}}
    />
    <p
      className="text-0-2-3 text-d0-0-2-4"
    >
      FÃ©licitation 
      choukri
      , vous avez bien inviter vos apprenants
    </p>
    <hr
      className="line-0-2-2"
    />
    <div
      className="row row-0-2-7 row-d2-0-2-11"
    >
      <div
        className="content_2"
      >
        <button
          className="button-0-2-13 button-d0-0-2-15"
          onClick={[Function]}
          type="button"
        >
           
          <p
            className="text-0-2-17 text-d0-0-2-18 text-0-2-14 text-d1-0-2-16"
          >
            Confirmer
          </p>
        </button>
      </div>
    </div>
  </div>
</div>
`;
